src/main/java/fr/insee/genesis/controller/adapter/LunaticXmlAdapter.java

```diff
@@ -13,21 +15,66 @@
+        SurveyUnitUpdateDto surveyUnitUpdateDto = getStateDataFromSurveyUnit(su, variablesMap, idCampaign, DataState.COLLECTED);
+
+        surveyUnitUpdateDtoList.add(surveyUnitUpdateDto);
+
+        //Get data from other states
+        SurveyUnitUpdateDto editedSurveyUnitUpdateDto = getStateDataFromSurveyUnit(su, variablesMap, idCampaign, DataState.EDITED);
+        if(editedSurveyUnitUpdateDto != null) surveyUnitUpdateDtoList.add(editedSurveyUnitUpdateDto);
+
+        SurveyUnitUpdateDto inputedSurveyUnitUpdateDto = getStateDataFromSurveyUnit(su, variablesMap, idCampaign, DataState.INPUTED);
+        if(inputedSurveyUnitUpdateDto != null) surveyUnitUpdateDtoList.add(inputedSurveyUnitUpdateDto);
+
+        SurveyUnitUpdateDto forcedSurveyUnitUpdateDto = getStateDataFromSurveyUnit(su, variablesMap, idCampaign, DataState.FORCED);
+        if(forcedSurveyUnitUpdateDto != null) surveyUnitUpdateDtoList.add(forcedSurveyUnitUpdateDto);
+
+        SurveyUnitUpdateDto previousSurveyUnitUpdateDto = getStateDataFromSurveyUnit(su, variablesMap, idCampaign, DataState.PREVIOUS);
+        if(previousSurveyUnitUpdateDto != null) surveyUnitUpdateDtoList.add(previousSurveyUnitUpdateDto);
+
+
+        return surveyUnitUpdateDtoList;
+    }
+
+    /**
+     * Collects data from XML survey unit depending on the data state
+     * @param su source XML Survey Unit
+     * @param variablesMap variable definitions (used for loops)
+     * @param idCampaign survey id
+     * @param dataState state of the DTO to generate
+     * @return Survey Unit DTO with a specific state
+     */
+    private static SurveyUnitUpdateDto getStateDataFromSurveyUnit(LunaticXmlSurveyUnit su, VariablesMap variablesMap, String idCampaign, DataState dataState) {
         SurveyUnitUpdateDto surveyUnitUpdateDto = SurveyUnitUpdateDto.builder()
                 .idQuest(su.getQuestionnaireModelId())
                 .idCampaign(idCampaign)
                 .idUE(su.getId())
-                .state(DataState.COLLECTED)
+                .state(dataState)
                 .mode(Mode.WEB)
                 .recordDate(LocalDateTime.now())
                 .fileDate(su.getFileDate())
                 .build();
-        List<CollectedVariableDto> variablesUpdate = new ArrayList<>();
 
-        /*For now, we don't manage list of values, we always define a list of one element in values
-        * To be updated when we will receive data from dynamic tables*/
-        su.getData().getCollected().forEach(lunaticXmlCollectedData -> {
+        if(dataState.equals(DataState.COLLECTED))
+            return getCollectedDataFromSurveyUnit(su, surveyUnitUpdateDto, variablesMap);
+        else{
+            return getOtherDataFromSurveyUnit(su, surveyUnitUpdateDto, variablesMap,dataState);
+        }
+    }
+
+    private static SurveyUnitUpdateDto getCollectedDataFromSurveyUnit(LunaticXmlSurveyUnit su, SurveyUnitUpdateDto surveyUnitUpdateDto, VariablesMap variablesMap) {
```

Comment:
This method should only treat External data. The Collected data with Collected state should be treated in the next method (see my next comment).